### PR TITLE
simplify Maryam file 

### PR DESCRIPTION
**Pull request by rishi23root**
1. make the Maryam file run only when calling from the command line
2. remove the use of v variable in function scope, bug is not visible due to [closure in python](https://www.geeksforgeeks.org/python-closures/)
3. error - "No value for argument 'line' in unbound method call from inline 42" in Maryam file, fix by changing the name of the function variable x - X
4. add __init__.py files to create packages from folders

### CODE REVIEW

**Comment by saeeddhqan** [comment]
I think you didn't use the last release. Maryam and __init__ files are acceptable. remove the rest.

File: core/core.py [hunk 1]
```diff
@@ -986,6 +986,6 @@ def complete_history(self, text, line, begidx, endidx):
 		return [x for x in ['off', 'on', 'status', 'all', 'from ', 'clear'] if x.startswith(text.lower())]
 
 	def complete_show(self, text, line, begidx, endidx):
-		args = line.split()
+		# args = line.split()
```
Comment (saeeddhqan) [inline core/core.py:989]: Remove it

File: core/core.py [hunk 2]
```diff
@@ -22,11 +19,14 @@
 import subprocess
 import sys
 import traceback
-import requests
-from core.util import rand_uagent
 from io import StringIO
 from textwrap import wrap
 
+import requests
```
Comment (saeeddhqan) [inline core/core.py:25]: no

File: core/core.py [hunk 3]
```diff
@@ -22,11 +19,14 @@
...
+
+from core.util import rand_uagent
```
Comment (saeeddhqan) [inline core/core.py:27]: no

**Comment by saeeddhqan** [comment]
@z3y50n check the Maryam file and let me know

File: core/__init__.py [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# for creating a directory as a package
+# more - https://docs.python.org/3/reference/import.html
```
Comment (z3y50n) [inline core/__init__.py:2]: This is not necessary in my opinion.

File: core/core.py [hunk 5]
```diff
@@ -616,7 +616,7 @@ def do_set(self, params):
 		name = options[0].lower()
 		if name in self._global_options:
 			value = ' '.join(options[1:])
-			if self._global_options_[name][2] and not value:
+			if self._global_options_[name][2] and (not value or value == 'None'):
```
Comment (z3y50n) [inline core/core.py:619]: Is `value == 'None'` necessary?

File: core/util/__init__.py [hunk 6]
```diff
@@ -0,0 +1,2 @@
+# for creating a directory as a package
+# more - https://docs.python.org/3/reference/import.html
```
Comment (z3y50n) [inline core/util/__init__.py:2]: Still unnecessary

File: maryam [hunk 7]
```diff
@@ -41,37 +32,45 @@ def cui(args, format_help):
 		if option in ['-v', '--version']:
 			print(f"OWASP MARYAM V.{__VERSION__}")
 			exit(0)
-		command = ' '.join(args[1:])
-		if not v or option not in ['-e', '--execute', '-s', '--start']:
+		if not option or option not in ['-e', '--execute', '-s', '--start']:
```
Comment (z3y50n) [inline maryam:35]: I think checking for `not option` is redundant since inside this `if` there will always be arguments

File: maryam [hunk 8]
```diff
@@ -41,37 +32,45 @@ def cui(args, format_help):
...
-		x = x('execute')
-		if option in ['-e', '--execute']:
-			x.onecmd(command)
-			exit(0)
-		elif args[0] in ['-s', '--start']:
-			x.onecmd(command)
-	else:
-		x = x('run')
-	while True:
+		X = x('execute')
```
Comment (z3y50n) [inline maryam:38]: Uppercase letter should be used only for data structures like trees

File: maryam [hunk 9]
```diff
@@ -41,37 +32,45 @@ def cui(args, format_help):
...
+		if option in ['-e', '--execute','-s', '--start']:
+			# passing command --> ' '.join(args[1:])
+			X.onecmd(' '.join(args[1:]))
+			if option in ['-e', '--execute']:
+				exit(0)
+	else :
```
Comment (z3y50n) [inline maryam:44]: don't put space before `:`

File: maryam [hunk 10]
```diff
@@ -41,37 +32,45 @@ def cui(args, format_help):
...
-	while True:
+		X = x('execute')
+		if option in ['-e', '--execute','-s', '--start']:
+			# passing command --> ' '.join(args[1:])
+			X.onecmd(' '.join(args[1:]))
+			if option in ['-e', '--execute']:
+				exit(0)
+	else : 
+		X = x('run')
+	while True :
```
Comment (z3y50n) [inline maryam:46]: don't put space before `:`

**Comment by z3y50n** [comment]
No major changes in this PR, I tested the tool and it works the same way.

File: core/core.py (hunk 5 above)
Comment (saeeddhqan) [reply core/core.py:619]: It is now part of the core.

**Pull request closed without merging**